var/spack/repos/builtin/packages/amdscalapack/package.py

```diff
@@ -38,11 +39,13 @@ def cmake_args(self):
         args = super(Amdscalapack, self).cmake_args()
         spec = self.spec
 
-        args.extend([
-            "-DUSE_DOTC_WRAPPER:BOOL=%s" % (
-                'ON' if '%aocc ^amdblis' in spec else 'OFF'
-            )
-        ])
+        if spec.satisfies('%gcc@10:'):
+            args.extend(['-DCMAKE_Fortran_FLAGS={0}'.format(
+                        "-fallow-argument-mismatch")])
+
+        if spec.satisfies('@2.2'):
```

Comment:
Does this argument only apply to version `2.2`?  If not, you might want to use a range check here.  

https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=spec.satisfies#testing-spec-constraints

Reply:
Yes, this argument only apply to 2.2 version.